lib/utils/aws/endpoint/endpoint.go

```diff
@@ -0,0 +1,29 @@
+package endpoint
+
+import (
+	"fmt"
+	"regexp"
+)
+
+var schemeRegex = regexp.MustCompile("^([^:]+)://")
```

Comment:
nit: Do we really have to write our own regular expression? There is no function in the standard packages doing that?